### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by shosuke-13**
### **PR Type**
enhancement, tests

___

### **Description**
- Updated the prediction logic in `train.py` to correctly handle binary classification tasks.
- Added new shell scripts for running experiments on various genomic tasks including lncrna, splicing, and others.
- Configured multiple nucleotide-transformer models and set training parameters such as learning rate, batch size, and epochs.
- Implemented loops in scripts to iterate over different seeds for robustness in experiments.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>train.py</strong><dd><code>Update prediction logic for binary classification tasks</code>&nbsp; &nbsp; </dd></summary>
<hr>

finetune/train.py

<li>Changed the condition for determining predictions from <br>"classification" to "binary_classification".<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/4/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>exp2_lncrna.sh</strong><dd><code>Add script for lncrna task experiments</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_lncrna.sh

<li>Added a new script for running experiments on lncrna tasks.<br> <li> Configured multiple nucleotide-transformer models for training.<br> <li> Set various training parameters including learning rate and batch <br>size.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/4/files#diff-c240a01021bd80f0495797ada58292c790a9cafc20bd39d705032f43b960230c">+39/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_splicing.sh</strong><dd><code>Add script for splicing task experiments</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_splicing.sh

<li>Added a new script for running experiments on splicing tasks.<br> <li> Configured multiple nucleotide-transformer models for training.<br> <li> Set various training parameters including learning rate and batch <br>size.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/4/files#diff-d96b1e011c3aa1986b78bd19f90d62f0f30f1f9df30dee3a0a4e3ab28332c873">+39/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exp2_tasks.sh</strong><dd><code>Add script for multiple genomic task experiments</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

finetune/scripts/dok/exp2_tasks.sh

<li>Added a new script for running experiments on multiple genomic tasks.<br> <li> Configured multiple nucleotide-transformer models for training.<br> <li> Implemented a loop to iterate over different seeds for robustness.<br>

</details>

  </td>
  <td><a href="https://github.com/shosuke-13/kmer-tokenization-strategy/pull/4/files#diff-91a001f3c05a5d11ce7eaa70d58a4b808784a40e4001a69ead955d9e8efc0cad">+52/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by shosuke-13** [comment]
Preparing PR description...

**Comment by shosuke-13** [comment]
Preparing review...

**Comment by shosuke-13** [comment]
## PR Reviewer Guide 🔍

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 4 🔵🔵🔵🔵⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Key issues to review</strong><br><br>

<a href='https://github.com/shosuke-13/kmer-tokenization-strategy/pull/4/files#diff-85b2ae89408b47a70c8c959c736279b2e68cc6ad331ab8f99c05e4fa8ee57016R222-R222'><strong>Logic Change</strong></a><br>The condition for determining predictions has changed from "classification" to "binary_classification". Ensure that this change does not introduce any unintended side effects in the prediction logic.

</td></tr>
</table>

**Comment by shosuke-13** [comment]
## Generating PR code suggestions


Work in progress ...<br>
